### PR TITLE
[gedi] Improvements based on updated extractor in yt-dl

### PR DESCRIPTION
**Pull request by nixxo**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

the gedi extractor in this fork is my old extractor. This is the updated version.

### CODE REVIEW

File: yt_dlp/extractor/gedidigital.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                                r'-video-rrtv-(\d+)', v, 'abr', default=None))
+                    formats.append(f)
+            elif t == 'param':
+                if n in ['image_full', 'image']:
+                    thumb = v
+                elif n == 'videoDuration':
+                    duration = int_or_none(v)
+
+        self._sort_formats(formats)
+        self._clean_formats(formats)
```
Comment (pukkandan) [inline yt_dlp/extractor/gedidigital.py:200]:
```suggestion
        self._clean_formats(formats)
        self._sort_formats(formats)
```

Unless there is a specific reason to do so, the format dict should not be modified after sorting

File: yt_dlp/extractor/gedidigital.py [hunk 2]
```diff
@@ -94,9 +101,49 @@ class GediDigitalIE(InfoExtractor):
...
+        return urls[0] if urls else None
+
+    @staticmethod
+    def _clean_formats(formats):
+        format_urls = set()
+        clean_formats = []
+        for f in formats:
+            if f['url'] not in format_urls:
+                if f.get('audio_ext') != 'none' and not f.get('acodec'):
+                    continue
```
Comment (pukkandan) [inline yt_dlp/extractor/gedidigital.py:139]: could you explain the intention behind this?

Comment (nixxo) [reply yt_dlp/extractor/gedidigital.py:139]: remove duplicate audio formats generated by the _extract_m3u8_formats method.


```
audio-hls-vod-1@48000-64000-Audio_64 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-1@48000-80000-Audio_80 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-2@48000-96000-Audio_96 mp4 audio only |       m3u8_n |                   unknown        
audio-hls-vod-100                    mp4 audio only |  100k m3u8_n |                   mp4a.40.2 100k 
audio-hls-vod-116                    mp4 audio only |  116k m3u8_n |                   mp4a.40.2 116k 
audio-hls-vod-133                    mp4 audio only |  133k m3u8_n |                   mp4a.40.2 133k 
```

as you can see there are 6 audio formats but only 3 in reality. I removed the 3 formats that have a "weird" `format_id` and missing `acodec` / `bitrate` information

Comment (pukkandan) [reply yt_dlp/extractor/gedidigital.py:139]: You are not checking whether the "more detailed" format exists though. I am gonna assume u are sure that such a format always exists

Comment (nixxo) [reply yt_dlp/extractor/gedidigital.py:139]: yup.

it's the _extract_m3u8_formats method that IMHO is designed with video streams and for only audio formats it does not check well and create duplicates.

**Pull request merged**